examples/gno.land/p/demo/queue/queue.gno

```diff
@@ -0,0 +1,118 @@
+package queue
+
+// Queue is a queue data structure that stores elements of any type.
+type Queue struct {
+	begin uint64                 // Index of the front element
+	end   uint64                 // Index after the last element
+	data  map[uint64]interface{} // Storage for queue elements
+}
+
+// NewQueue creates and initializes a new queue.
+func NewQueue() *Queue {
+	return &Queue{
+		begin: 0,
+		end:   0,
+		data:  make(map[uint64]interface{}),
+	}
+}
+
+// PushBack adds an element to the end of the queue.
+// Returns an error if the queue is full.
+func (q *Queue) PushBack(value interface{}) error {
+	if q.end+1 == q.begin {
+		return ErrResourceError
+	}
+
+	q.data[q.end] = value
+	q.end++
+
+	return nil
+}
+
+// PopBack removes and returns the element at the end of the queue.
+// Returns an error if the queue is empty.
+func (q *Queue) PopBack() (interface{}, error) {
+	if q.begin == q.end {
+		return nil, ErrEmptyArrayPop
+	}
+
+	q.end--
+	value := q.data[q.end]
+	delete(q.data, q.end)
+
+	return value, nil
+}
+
+// PushFront adds an element to the front of the queue.
+// Returns an error if the queue is full.
+func (q *Queue) PushFront(value interface{}) error {
+	if q.begin-1 == q.end {
```

Comment:
With `q.begin` set to 0 by default, you may encounter an issue here.

Reply:
I have tested this case, `q.begin` will equal `max.Uint64`